src/extensions/lens-main-extension.ts

```diff
@@ -4,42 +4,55 @@
  */
 
 import { LensExtension } from "./lens-extension";
-import { WindowManager } from "../main/window-manager";
-import { catalogEntityRegistry } from "../main/catalog";
 import type { CatalogEntity } from "../common/catalog";
 import type { IObservableArray } from "mobx";
 import type { MenuRegistration } from "../main/menu/menu-registration";
 import type { TrayMenuRegistration } from "../main/tray/tray-menu-registration";
 import type { ShellEnvModifier } from "../main/shell-session/shell-env-modifier/shell-env-modifier-registration";
+import { Environments, getEnvironmentSpecificLegacyGlobalDiForExtensionApi } from "./as-legacy-globals-for-extension-api/legacy-global-di-for-extension-api";
+import catalogEntityRegistryInjectable from "../main/catalog/catalog-entity-registry.injectable";
+import navigateForExtensionInjectable from "../main/start-main-application/lens-window/navigate-for-extension.injectable";
 
 export class LensMainExtension extends LensExtension {
   appMenus: MenuRegistration[] = [];
   trayMenus: TrayMenuRegistration[] = [];
 
   /**
    * implement this to modify the shell environment that Lens terminals are opened with. The ShellEnvModifier type has the signature
-   * 
+   *
    * (ctx: ShellEnvContext, env: Record<string, string | undefined>) => Record<string, string | undefined>
-   * 
+   *
    *  @param ctx the shell environment context, specifically the relevant catalog entity for the terminal. This can be used, for example, to get
    * cluster-specific information that can be made available in the shell environment by the implementation of terminalShellEnvModifier
-   * 
+   *
    * @param env the current shell environment that the terminal will be opened with. The implementation should modify this as desired.
-   * 
+   *
    * @returns the modified shell environment that the terminal will be opened with. The implementation must return env as passed in, if it
    * does not modify the shell environment
    */
   terminalShellEnvModifier?: ShellEnvModifier;
 
   async navigate(pageId?: string, params?: Record<string, any>, frameId?: number) {
-    return WindowManager.getInstance().navigateExtension(this.id, pageId, params, frameId);
+    const di = getEnvironmentSpecificLegacyGlobalDiForExtensionApi(Environments.main);
```

Comment:
Why do it this way instead of changing the dependencies?

Reply:
Bigger thing to reorganize it in a way that we have full control over constructor of these abstract classes. Easy way to introduce unwanted breaking changes. Could be done, but we decided to keep it out from this PR.

Reply:
I can see that you have done the work required to get dependencies for these abstract classes. So doing this after branches have been merged.

Reply:
Changed to use dependencies.